mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -1357,6 +1372,26 @@ impl TendermintCoin {
     }
 }
 
+fn clients_from_urls(rpc_urls: &[String]) -> MmResult<Vec<HttpClient>, TendermintInitErrorKind> {
+    let mut clients = Vec::new();
+    let mut errors = Vec::new();
+    // check that all urls are valid
+    // keep all invalid urls in one vector to show all of them in error
+    for url in rpc_urls.iter() {
+        match HttpClient::new(url.as_str()) {
+            Ok(client) => clients.push(client),
+            Err(e) => errors.push(format!("Url {} is invalid, got error {}", url, e)),
+        }
+    }
+    drop_mutability!(clients);
+    drop_mutability!(errors);
+    if !errors.is_empty() {
+        let errors: String = errors.into_iter().join(", ");
+        return MmError::err(TendermintInitErrorKind::RpcClientInitError(errors));
+    }
+    Ok(clients)
```

Comment:
If `rpc_urls.is_empty()`, `errors` will also be empty. Do we need to return an error if `rpc_urls.is_empty()`? 🤔

Reply:
well, we check if `rpc_urls` is empty [here](https://github.com/KomodoPlatform/atomicDEX-API/blob/tendermint-multiple-rpcs-optimization/mm2src/coins/tendermint/tendermint_coin.rs#L436). This check is performed at the very beginning before calling `clients_from_urls` function, so `rpc_urls` should have any urls.
But why not, I will add one more check at the beginning of `clients_from_urls` function, just to avoid unpredictable situations (like we do with `drop_mutability!` macro).
Done